### PR TITLE
Heedls 997 Fix add new course journey retain values

### PR DESCRIPTION
**Pull request by VladNicol**
### JIRA link
[HEEDLS-997](https://softwiretech.atlassian.net/browse/HEEDLS-997)

### Description
Alex fixed a bug where on the add course journey the values selected on the first page would not be preserved when clicking next and then going back.

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [ ] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [ ] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/TrackingSystem/CourseSetup/AddNewCentreCourse/_SelectCourseFilterableSelectListItem.cshtml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 ﻿@using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.AddNewCentreCourse
 @model FilterableApplicationSelectListItemViewModel
 
-<option value="@Model.ApplicationId.ToString()"
-        class="searchable-element searchable-element-title"
-        category-filter-value="@Model.CategoryFilter"
-        topic-filter-value="@Model.TopicFilter">
+@{ var selected = Model.Selected ? @"selected=""selected""" : ""; }
+
+<!option value="@Model.ApplicationId.ToString()"
```
Comment (livzorn) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/CourseSetup/AddNewCentreCourse/_SelectCourseFilterableSelectListItem.cshtml:6]: I haven't seen an `!` used in an HTML tag before, except for with DOCTYPE. What does it do?

**Review by livzorn — APPROVED** [review]
Looks good, just one question for my own curiosity.

File: DigitalLearningSolutions.Web/Views/TrackingSystem/CourseSetup/AddNewCentreCourse/_SelectCourseFilterableSelectListItem.cshtml (hunk 1 above)
Comment (VladNicol) [reply DigitalLearningSolutions.Web/Views/TrackingSystem/CourseSetup/AddNewCentreCourse/_SelectCourseFilterableSelectListItem.cshtml]: I don't know honestly. I tried googling this but I didn't find anything useful. One difference that I noticed is that without it I get an error saying `The tag helper 'option' must not have C# in the element's attribute declaration area`, so we couldn't use `@selected` as an attribute. I pushed a work around to this that seems to have the same functionality, but without the `!`, I think it's better than having things we don't understand.

@SteveJacksonSoft  or maybe @simonbrent Can you explain what `!` does in an HTML tag?

**Review by simonbrent — APPROVED** [review]
Looks good to me

**Pull request merged**